File: Python/Program/CatOwner.py
class CatOwner:
    def __init__(self, owner_name):
        self.owner_name = owner_name
        self.cats = []

    def add_cat(self, cat):
        self.cats.append(cat)

    def show_cats(self):
        print(f"{self.owner_name}'s Pet Cats:")
        if not self.cats:
            print("No cats found.")
        for i, cat in enumerate(self.cats, 1):
            print(f"{i}. ", end="")
            cat.display()
            print()

    def display(self):
        print(f"Owner Name: {self.owner_name}")
        self.show_cats()

File: CPP/Program/CatOwner.cpp
#pragma once
#include "PetCat.cpp"
#include <vector>

class CatOwner {
    string ownerName;
    vector<PetCat *> cats;

public:
    CatOwner(const string &name) : ownerName(name) {}

    void addCat(PetCat *cat) {
        cats.push_back(cat);
    }

    void showCats() const {
        cout << ownerName << "'s Pet Cats: " << endl;
        for (int i = 0; i < cats.size(); i++) {
            cout << i + 1 << ". ";
            cats[i]->display();
            cout << endl;
        }
        if (cats.size() == 0) {
            cout << "No cats found." << endl;
        }
    }

    void display() const {
        cout << "Owner Name: " << ownerName << endl;
        showCats();
    }

    ~CatOwner() {
        for (auto cat : cats) {
            delete cat;
        }
    }
};